File: problems/3005/solution.ex
defmodule Solution do
  @spec max_frequency_elements([integer]) :: integer
  def max_frequency_elements(nums) do
    freq = Enum.frequencies(nums)
    best = freq |> Map.values() |> Enum.max()
    freq |> Map.values() |> Enum.filter(&(&1 == best)) |> Enum.sum()
  end
end

File: problems/3005/Solution.kt
class Solution {
    fun maxFrequencyElements(nums: IntArray): Int {
        val freq = IntArray(101)
        var best = 0
        for (x in nums) {
            freq[x]++
            best = maxOf(best, freq[x])
        }
        return freq.filter { it == best }.sum()
    }
}
